main/test/ca/uwaterloo/flix/library/Pkger/TestParser.flix

```diff
@@ -0,0 +1,454 @@
+// seperatedTake                                                                             //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def seperatedTake1(): Bool =
+    String.toList("a.b.c") |> seperatedTake(char(), tag('.'), 2)
+    ==
+    Some(('.' :: 'c' :: Nil, 'a' :: 'b' :: Nil))
+   
+@test
+def seperatedTake2(): Bool =
+    String.toList("a.") |> seperatedTake(char(), tag('.'), 2)
+    ==
+    None
+   
+@test
+def seperatedTake3(): Bool =
+    String.toList(".a") |> seperatedTake(char(), tag('.'), 0)
+    ==
+    Some(('.' :: 'a' :: Nil, Nil))
+   
+@test
+def seperatedTake4(): Bool =
+    String.toList(".a") |> seperatedTake(char(), tag('.'), -1)
+    ==
+    Some(('.' :: 'a' :: Nil, Nil))
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+// tailSeperatedTake                                                                         //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def tailSeperatedTake1(): Bool =
+    String.toList("a.b.c") |> tailSeperatedTake(char(), tag('.'), 2)
+    ==
+    Some(('c' :: Nil, 'a' :: 'b' :: Nil))
+   
+@test
+def tailSeperatedTake2(): Bool =
+    String.toList("a.") |> tailSeperatedTake(char(), tag('.'), 2) == None
+   
+@test
+def tailSeperatedTake3(): Bool =
+    String.toList(".a") |> tailSeperatedTake(char(), tag('.'), 0) == Some(('a' :: Nil, Nil))
+   
+@test
+def tailSeperatedTake4(): Bool =
+    String.toList(".a") |> tailSeperatedTake(char(), tag('.'), -1) == Some(('a' :: Nil, Nil))
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+// takeAny                                                                                   //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def takeAny1(): Bool =
+    Nil |> takeAny(tag('a')) == Some((Nil, Nil))
+
+@test
+def takeAny2(): Bool =
+    ('a' :: Nil) |> takeAny(tag('a')) == Some((Nil, 'a' :: Nil))
+
+@test
+def takeAny3(): Bool = 
+    ('a' :: 'b' :: Nil) |> takeAny(char().filter(Char.isLetter))
+    ==
+    Some((Nil, 'a' :: 'b' :: Nil))
+
+@test
+def takeAny4(): Bool = 
+
+    ('a' :: 'b' :: '-' :: Nil) |> takeAny(char().filter(Char.isLetter))
+    ==
+    Some(('-' :: Nil, 'a' :: 'b' :: Nil))
+
+@test
+def takeAny5(): Bool = 
+
+    ('a' :: 'b' :: '-' :: 'c' :: Nil) |> takeAny(char().filter(Char.isLetter))
+    ==
+    Some(('-' :: 'c' :: Nil, 'a' :: 'b' :: Nil))
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+// takeMany                                                                                  //
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+@test
+def takeMany1(): Bool = 
+    ('a' :: 'b' :: Nil) |> takeMany(char().filter(Char.isLetter))
```

Comment:
It is inconsistent whether you use `String.toList` or you write the literal list. I would prefer the literal list, like you do here.

Reply:
I've changed anything that's less than 3 characters to use list notation, after that it becomes more verbose.